### ISSUE TITLE
Is there a way to remove this warning - 

### ISSUE THREAD

**Issue by VIGNESHinZONE** [materialsvirtuallab/megnet#233]
As i run this piece of code, I get a warning message

```
model = MEGNetModel(100, 2,
                    graph_converter=CrystalGraph(bond_converter=GaussianDistance(np.linspace(0, 5, 100), 0.5)))

model.train(train, target, epochs=1)
```

Here is the warning message I get - 
```
/usr/local/lib/python3.7/dist-packages/tensorflow/python/framework/indexed_slices.py:437: UserWarning: Converting sparse IndexedSlices(IndexedSlices(indices=Tensor("gradient_tape/model_1/set2set_atom/Reshape_9:0", shape=(None,), dtype=int32), values=Tensor("gradient_tape/model_1/set2set_atom/Reshape_8:0", shape=(None,), dtype=float32), dense_shape=Tensor("gradient_tape/model_1/set2set_atom/Cast:0", shape=(1,), dtype=int32))) to a dense Tensor of unknown shape. This may consume a large amount of memory.
  "shape. This may consume a large amount of memory." % value)
```

We are trying to build a wrapper function for the megnet model and it's important for us to remove this error.

![Screenshot from 2021-03-18 17-23-25](https://user-images.githubusercontent.com/45672459/111622200-e4887580-880e-11eb-9b25-e10911ef3767.png)

**Comment by chc273** [comment]
@VIGNESHinZONE This was caused by the gradient calculations on `tf.gather` ops. Please see if the latest release `v.1.2.7` fix this issue.

**Comment by VIGNESHinZONE** [comment]
Yes it did, Thank You very much

**Comment by VIGNESHinZONE** [comment]
Sorry for reopening this issue @chc273, 

As I was noticing, is there any multitask classification output layer, which uses a softmax output layer? Please correct me if I'm wrong, but in the current version only binary classification tasks are available

**Comment by chc273** [comment]
@VIGNESHinZONE I have decided to revert back to the previous implementations (with warnings) for the new v1.2.8. In case you do not like the warning, you can keep using v1.2.7. I am changing it back since the one without warning is substantially slower. 

For multitask classification, you can do some simple modifications to the code and swap the activation function of the last layer when building the model.

**Comment by chc273** [comment]
Note you also need to change the loss function etc most likely.
